src/code/z_skin_matrix.c

```diff
@@ -1,42 +1,692 @@
+}
+// output matrix scales, then rotates (YRP), then translates the vector
+void SkinMatrix_ScaleRotateYRPTranslate(MtxF* mf, f32 scaleX, f32 scaleY, f32 scaleZ, s16 yaw, s16 roll, s16 pitch,
+                                        f32 dx, f32 dy, f32 dz) {
+    MtxF mft1;
+    MtxF mft2;
 
+    SkinMatrix_Translate(mf, dx, dy, dz);
+    SkinMatrix_RotateYRP(&mft1, yaw, roll, pitch);
+    SkinMatrix_MtxFMtxFMult(mf, &mft1, &mft2);
+    SkinMatrix_Scale(&mft1, scaleX, scaleY, scaleZ);
+    SkinMatrix_MtxFMtxFMult(&mft2, &mft1, mf);
+}
+
+// output matrix rotates (YRP), then translates the vector
+void SkinMatrix_RotateRPYTranslate(MtxF* mf, s16 roll, s16 pitch, s16 yaw, f32 dx, f32 dy, f32 dz) {
+    MtxF mft1;
+    MtxF mft2;
+
+    SkinMatrix_Translate(&mft2, dx, dy, dz);
+    SkinMatrix_RotateRPY(&mft1, roll, pitch, yaw);
+    SkinMatrix_MtxFMtxFMult(&mft2, &mft1, mf);
+}
+
+void SkinMatrix_Vec3fToVec3s(Vec3f* src, Vec3s* dest) {
+    dest->x = src->x;
+    dest->y = src->y;
+    dest->z = src->z;
+}
+
+// SkinMatrix_Vec3sToVec3f
+void SkinMatrix_Vec3sToVec3f(Vec3s* src, Vec3f* dest) {
+    dest->x = src->x;
+    dest->y = src->y;
+    dest->z = src->z;
+}
+
+// SkinMatrix_MtxFToMtx
+void SkinMatrix_MtxFToMtx(MtxF* src, Mtx* dest) {
+    s32 temp;
+    u16* m1 = (u16*)&dest->m[0][0];
+    u16* m2 = (u16*)&dest->m[2][0];
+
+    temp = src->xx * 0x10000;
+    m1[0] = (temp >> 0x10);
+    m1[16 + 0] = temp & 0xFFFF;
+
+    temp = src->xy * 0x10000;
+    m1[1] = (temp >> 0x10);
+    m1[16 + 1] = temp & 0xFFFF;
+
+    temp = src->xz * 0x10000;
+    m1[2] = (temp >> 0x10);
+    m1[16 + 2] = temp & 0xFFFF;
+
+    temp = src->xw * 0x10000;
+    m1[3] = (temp >> 0x10);
+    m1[16 + 3] = temp & 0xFFFF;
+
+    temp = src->yx * 0x10000;
+    m1[4] = (temp >> 0x10);
+    m1[16 + 4] = temp & 0xFFFF;
+
+    temp = src->yy * 0x10000;
+    m1[5] = (temp >> 0x10);
+    m1[16 + 5] = temp & 0xFFFF;
+
+    temp = src->yz * 0x10000;
+    m1[6] = (temp >> 0x10);
+    m1[16 + 6] = temp & 0xFFFF;
+
+    temp = src->yw * 0x10000;
+    m1[7] = (temp >> 0x10);
+    m1[16 + 7] = temp & 0xFFFF;
+
+    temp = src->zx * 0x10000;
+    m1[8] = (temp >> 0x10);
+    m1[16 + 8] = temp & 0xFFFF;
+
+    temp = src->zy * 0x10000;
+    m1[9] = (temp >> 0x10);
+    m2[9] = temp & 0xFFFF;
+
+    temp = src->zz * 0x10000;
+    m1[10] = (temp >> 0x10);
+    m2[10] = temp & 0xFFFF;
+
+    temp = src->zw * 0x10000;
+    m1[11] = (temp >> 0x10);
+    m2[11] = temp & 0xFFFF;
+
+    temp = src->wx * 0x10000;
+    m1[12] = (temp >> 0x10);
+    m2[12] = temp & 0xFFFF;
+
+    temp = src->wy * 0x10000;
+    m1[13] = (temp >> 0x10);
+    m2[13] = temp & 0xFFFF;
+
+    temp = src->wz * 0x10000;
+    m1[14] = (temp >> 0x10);
+    m2[14] = temp & 0xFFFF;
+
+    temp = src->ww * 0x10000;
+    m1[15] = (temp >> 0x10);
+    m2[15] = temp & 0xFFFF;
+}
+
+Mtx* func_800A7E70(GraphicsContext* gfxCtx, MtxF* src) {
+    Mtx* displayList = Graph_Alloc(gfxCtx, 0x40);
+
+    if (displayList == 0) {
```

Comment:
is this a dlist? if it was it wouldnt be of type Mtx*
ID also make the comparison below for NULL instead of 0